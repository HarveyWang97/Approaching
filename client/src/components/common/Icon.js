import React, { Component } from 'react';
import '../../assets/styles/Popup.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

/**
 * @classdesc Construct a FontAwesonIcon. 
 */
class Icon extends ComponentÂ {
    /**
	 * Render the icon with given iconName.
	 * @param {Function} onClick Associate specific onClick behavior with this icon.
     * @param {String} iconName The name of FAIcon to be constructed.
     * @returns {html} Returns an html object of FAIcon. 
	 */
    render() {
        const { onClick, iconName } = this.props;
        return (
            <span className='popup_icon' onClick={onClick}>
                <FontAwesomeIcon icon={iconName} className='fa-lg'/>
            </span>
        );
    }
}

export default Icon;
