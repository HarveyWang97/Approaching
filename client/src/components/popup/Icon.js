import React, { Component } from 'react';
import '../../css/Popup.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

/**
 * @classdesc Construct a FontAwesonIcon. 
 * @param {JsonObject} onClick Associate specific onClick behavior with this icon.
 * @param {string} iconName The name of FAIcon to be constructed.
 */
class Icon extends ComponentÂ {
    /**
	 * Render the icon with given iconName.
	 * @param {none}
     * @return {html} Return a html object of FAIcon. 
	 */
    render() {
        const { onClick, iconName } = this.props;
        return (
            <span className='popup_icon' onClick={onClick}>
                <FontAwesomeIcon icon={iconName} className='fa-lg'/>
            </span>
        );
    }
}

export default Icon;
