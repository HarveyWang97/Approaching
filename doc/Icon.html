<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Icon</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Icon</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Icon<span class="signature">(onClick, iconName)</span><span class="type-signature"></span></h2>
        
            <div class="class-description">Construct a FontAwesonIcon.</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="Icon"><span class="type-signature"></span>new Icon<span class="signature">(onClick, iconName)</span><span class="type-signature"></span></h4>
    

    











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>onClick</code></td>
            

            <td class="type">
            
                
<span class="param-type">JsonObject</span>


            
            </td>

            

            

            <td class="description last">Associate specific onClick behavior with this icon.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>iconName</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The name of FAIcon to be constructed.</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_popup_Icon.js.html">components/popup/Icon.js</a>, <a href="components_popup_Icon.js.html#line10">line 10</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="render"><span class="type-signature"></span>render<span class="signature">()</span><span class="type-signature"> &rarr; {html}</span></h4>
    

    



<div class="description">
    Render the icon with given iconName.
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            

            <td class="type">
            
                
<span class="param-type">none</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_popup_Icon.js.html">components/popup/Icon.js</a>, <a href="components_popup_Icon.js.html#line16">line 16</a>
    </li></ul></dd>
    

    

    

    
</dl>













<h5>Returns:</h5>

        
<div class="param-desc">
    Return a html object of FAIcon.
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">html</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Icon.html">Icon</a></li><li><a href="LoginButton.html">LoginButton</a></li><li><a href="module.exports.html">exports</a></li><li><a href="Popup.html">Popup</a></li><li><a href="Row.html">Row</a></li></ul><h3>Global</h3><ul><li><a href="global.html#changeEditingState">changeEditingState</a></li><li><a href="global.html#changeLayer">changeLayer</a></li><li><a href="global.html#fetchData">fetchData</a></li><li><a href="global.html#handleEditResult">handleEditResult</a></li><li><a href="global.html#handleSubmit">handleSubmit</a></li><li><a href="global.html#insertUser">insertUser</a></li><li><a href="global.html#renderItem">renderItem</a></li><li><a href="global.html#renderLayer">renderLayer</a></li><li><a href="global.html#renderMonth">renderMonth</a></li><li><a href="global.html#renderYear">renderYear</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Thu Nov 15 2018 14:21:25 GMT-0800 (Pacific Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>