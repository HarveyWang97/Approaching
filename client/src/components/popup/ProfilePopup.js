import React, { Component } from 'react';
import '../../css/Profile.css';
import Row from './Row';
import Icon from './Icon';

class Profile extends ComponentÂ {
    constructor(props){
        super(props);

        this.fieldToIcon = {
        	reminder: 'bell',
        	name: 'user',
            email: 'envelope'
            
        };
        this.data = {
            editing: false,
            data: {
                title: 'Settings',
                name: 'Joe Bruin',
                email: 'JoeBruin@ucla.edu',                
                reminder: '24 h'           
            }
        };

        this.state = {
            editing: this.data.editing,
        };
        for (let key in this.data.data) {
            this.state[key] = this.data.data[key];
        }
    }

    /**
	 * The changeEditingState() changes the current editing state to another.
	 * either from editing to not-editing, or vice versa.
	 *
	 * @param {None} 
	 * @return {void} 
	 *
	 * @example
	 *
	 *     changeEditingState()
	 */

    changeEditingState(){
        this.setState({
            editing: !this.state.editing
        });
    }

    /**
	 * The handleSubmit() invokes the changeEditingState() on user clicks and logs the change.
	 *
	 * @param {None}
	 * @return {void} 
	 *
	 * @example
	 *
	 *     handleSubmit()
	 */

    handleSubmit() {
        this.changeEditingState();
    }

    /**
	 * The handleEditResult(key, value) sets the data field of the state with the key = "key" to have value = "value"
	 * @param {String| String} key value
	 * @return {None} 
	 *
	 * @example
	 *
	 *     handleEditResult(key, value)
	 */
    handleEditResult(key, value) {
        this.state[key] = value;
    }

    render() {
        const data = this.data.data;
        return (
            <div className='profile'>
                <div className='profile_inner'>
                    <div className='top'>
                        <div>
                            <Icon iconName='times' onClick={this.props.closePopup} />
                            <div className='profile_title'>{this.state.title}</div>
                        </div>
                    </div>
                    <div className='middle'>
                        {Object.keys(data).map(key => {
                            if (key !== 'title') {
                                return (
                                    <Row key={key} 
                                         field={key}
                                         iconName={this.fieldToIcon[key]}
                                         details={this.state[key]}
                                         editing={this.state.editing}
                                         handleEditResult={this.handleEditResult.bind(this)} />
                                );
                            }
                        })}
                    </div>
                    <div className='bottom'>
                        <div className='left'>
                            { this.state.editing ? 
                            (<Icon iconName='save' onClick={this.handleSubmit.bind(this)} />)
                            : (<Icon iconName='pen' onClick={this.changeEditingState.bind(this)}/>)
                            }
                        </div>
                        <div className='right'>
                            <Icon iconName='sign-out-alt' />
                        </div>
                    </div>
                </div>
            </div>
        ); 
    }
}

export default Profile;



