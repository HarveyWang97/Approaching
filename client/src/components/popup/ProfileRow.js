import React, { Component } from 'react';
import config from '../../config';
import '../../css/Profile.css';
import ls from 'local-storage';
import Icon from './Icon';
import { lstat } from 'fs';

/**
 * @classdesc Called by Popup to construct a pair of one Icon and one text value. 
 * 
 */
class ProfileRow extends Component {
    /**
	 * Currently we manually construct datas for popup to display since we do not have communication with others.
     * Initialize the state variables with corresponding input data.
     * (TO BE DONE) Communication with other front-end components and server.
	 * @constructor
	 * @param {None}
	 * @return {void} 
	 */
    constructor(props) {
        super(props);

        this.state = {
            item: this.props.details
        };
    }

    /**
	 * This method set the value of this row's item to the new input value.
	 * 
	 * @param {JsonObject} event a specific event that invokes this method, e.g. editing the iput form
	 * @return {void} 
	 */
    handleChange(event){
        this.setState({
            item: event.target.value
        });
        this.props.handleEditResult(this.props.field,event.target.value);
        ls.set(this.props.field,event.target.value)
    }

    /**
	 * Render the row based on the given input. 
     *
     *  @param {String} field The data type of given text value, e.g. location, time.
     * @param {String} iconName The name of FAIcon to be constructed.
     * @param {Boolean} editing Whether the Popup is in editing mode.
     * @param {JsonObject} handleEditResult Pass handleEditResult behavior from Popup to this object.
     * 
     * @return {html} Returns a html block of Popup component. 
	 */
    render() {
        const { iconName, placeholder, details, editing } = this.props;
        return (
            <div className='profile_row'>
                    <div className='profile_icon'>
                        <Icon iconName={iconName}/>
                    </div>
                
                    {editing ? (<input type="text" value={this.state.item} placeholder={placeholder}
                            onChange={this.handleChange.bind(this)} />)
                            : iconName === config.icons["reminder"] ? (<span>{details} hour(s)</span>) : (<span>{details}</span>)
                    }
                </div>
        );
        
    }
}

export default ProfileRow;