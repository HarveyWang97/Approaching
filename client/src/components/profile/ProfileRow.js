import React, { Component } from 'react';
import config from '../../config';
import '../../assets/styles/Profile.css';
import ls from 'local-storage';
import Icon from '../common/Icon';

/**
 * @classdesc This class renders a row in the Profile.
 */
class ProfileRow extends Component {
    /**
     * Initialize the state variables with corresponding input property data. 
     * 
	 * @constructor
	 * @param {Object} props
	 * @return {void} 
	 */
    constructor(props) {
        super(props);

        this.state = {
            item: this.props.details
        };
    }

    /**
     * If the component receives a new props object that is different from current one,
     * replace the name state with the name in the new props object.
     * @param {Object} nextProps The new props.
     */
    componentWillReceiveProps(nextProps){
        if(nextProps.details !== this.props.details){
            this.setState({item:nextProps.details});
        }
    }

    /**
	 * This method set the value of this row's item to the new input value.
	 * 
	 * @param {JsonObject} event a specific event that invokes this method, e.g. editing the iput form
	 * @return {void} 
	 */
    handleChange(event){
        this.setState({
            item: event.target.value
        });
        this.props.handleEditResult(this.props.field,event.target.value);
        ls.set(this.props.field,event.target.value)
    }

    /**
	 * Render the row in profile based on the given input. 
     * @param {Object} 
     * @return {html} Returns a html block of Profile component. 
	 */
    /**
	 * Render the row based on the given input. 
     * @param {None} 
     * 
     * @return {html} Returns a html block of Popup component. 
	 */
    render() {
        const { iconName, placeholder, details, editing } = this.props;
        return (
            <div className='profile_row'>
                    <div className='profile_icon'>
                        <Icon iconName={iconName}/>
                    </div>
                
                    {editing ? (<input type="text" value={this.state.item} placeholder={placeholder}
                            onChange={this.handleChange.bind(this)} />)
                            : iconName === config.icons["reminder"] ? (<span>{details} hour(s)</span>) : (<span>{details}</span>)
                    }
                </div>
        );
        
    }
}

export default ProfileRow;