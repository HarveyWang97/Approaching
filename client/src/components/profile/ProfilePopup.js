import React, { Component } from 'react';
import '../../assets/styles/Profile.css';
import ProfileRow from './ProfileRow';
import Icon from '../common/Icon';
import {connect} from 'react-redux';
import  * as actions from '../../actions';
import ls from 'local-storage';
import config from '../../config';
import {Link} from 'react-router-dom';


/**
 * @classdesc Construct a Profile component that renders given data object. 
 */
class Profile extends ComponentÂ {
    /**
     * Initialize the state variables used in constructing the Profile component. 
     * The first four are manually set as flag and constant. The last two are retrieved 
     * from this.prop to get user profile.
	 * @constructor 
	 * @param {Object} props The properties passed in when the component is 
     * constructed. The component which invokes the popup should pass a payload
     * via this.props.payload, which contains contentType, isAdd, id, and
     * currentLocation which is used to decide what content is used for 
     * rendering. id is used as a key to search the item or event list in reducers to
     * get the details of that item or event.
	 * @returns {void} 
	 */
    constructor(props){
        super(props);
        this.state = {
            editing: false,
            validEmail: true,
            validReminder: true,
            title: 'Account Settings',
            email: this.props.userProfile.email, 
            reminder: this.convertMstoHours(this.props.userProfile.notifyTime)
        };
    }

    /**
     * If the component receives a new props object that is different from current one,
     * replace the name state with the name in the new props object.
     * @param {Object} nextProps The new props.
     */
    componentWillReceiveProps(nextProps){
        if(nextProps.userProfile.email !== this.props.userProfile.email){
            this.setState({email:nextProps.userProfile.email});
        }

        if(nextProps.userProfile.notifyTime !== this.props.userProfile.notifyTime){
            this.setState({reminder:this.convertMstoHours(nextProps.userProfile.notifyTime)});
        }
    }

    /**
     * Override the React function to fetch user profile from local storage during didMount stage.
     * @return {void}
     */
    componentDidMount(){
        const facebookId = ls.get('facebookId');
        const accessToken = ls.get('accessToken');
        this.props.fetchProfile(facebookId,accessToken);
    }

    /**
	 * Validate the format of email from the user input.
	 *
	 * @param {String} email
	 * @return {Boolean} Returns whether the input email address is valid or not.
	 *
	 * @example
	 *
	 *     validateEmail("test@ucla.edu")
	 */
    validateEmail(email) {
        var re = /^(([^<>()\]\\.,;:\s@"]+(\.[^<>()\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        return re.test(String(email).toLowerCase());
    }

    /**
	 * Validate the format of reminder time from the user input.
	 *
	 * @param {String} time The reminder time
	 * @return {Boolean} Returns whether time is a non-negative floating number.
	 *
	 * @example
	 *
	 *     validateReminder("24.32")
	 */
    validateReminder(time){
        return (!isNaN(time)) && (parseFloat(time) >= 0);
    }

    /**
	 * The changeEditingState() changes the current editing state to another.
	 * either from editing to not-editing, or vice versa.
	 *
	 * @param {None} 
	 * @return {void} 
	 *
	 * @example
	 *
	 *     changeEditingState()
	 */
    changeEditingState(){
        this.setState(prevState => ({
            editing: !prevState.editing
        }));
    }

    /**
	 * The handleSubmit() invokes the changeEditingState() on user clicks and logs the change.
     * It also invokes the validation methods for email and reminder time and change the flag correspondingly.
	 *It also connects with the backend to send updated data.
     * 
	 * @param {None}
	 * @return {void} 
	 *
	 * @example
	 *
	 *     handleSubmit()
	 */

    handleSubmit() {
        const facebookId = ls.get('facebookId');
        const accessToken = ls.get('accessToken');

        let ve;
        let vr;
        // check validity of email input
        if (this.validateEmail(this.state.email))
        {
            ve = true;
            this.setState({validEmail: true});
        }
        else{
            ve = false;
            this.setState({validEmail: false});
        }

        if (this.validateReminder(this.state.reminder)){
            vr = true
            this.setState({validReminder: true});        
        }
        else{
            vr = false;
            this.setState({validReminder: false}); 
        }
        
        if (ve  && vr){
            this.props.updateEmail(this.state.email, facebookId, accessToken)
            .then(() => this.props.updateNotifyTime(this.convertHourstoMs(this.state.reminder).toString(), facebookId, accessToken))
            .then(() => this.props.fetchProfile(facebookId,accessToken))
            .then( () => {
                this.changeEditingState();
            });
            
        }
        
    }

    /**
	 * The handleEditResult(key, value) sets the data field of the state with the key = "key" to have value = "value"
	 * @param {String| String} key value
	 * @return {None} 
	 *
	 * @example
	 *
	 *     handleEditResult(key, value)
	 */
    handleEditResult(key, value) {
        if (key === "email"){
            this.setState({email: value});
        } 
        if (key === "reminder"){
            this.setState({reminder: value});
        }         
        
    }

    /**
	 * Convert reminder time in hours to milliseconds for communication with backend.
	 * @param {String} hour
	 * @return {Float} ms
	 *
	 * @example
	 *
	 *     convertHourstoMs("24");
	 */
    convertHourstoMs(hour){
        return parseFloat(hour) * 3600000
    }

    /**
	 * Convert reminder time in milliseconds to hours for communication with backend.
	 * @param {String} sec
	 * @return {Float} hours
	 *
	 * @example
	 *
	 *     convertMstoHours("3600000");
	 */
    convertMstoHours(sec){
        return parseFloat(sec) / 3600000
    }


    /**
	 * Log out current user, clear local storage of that user, and close the profile popup.
	 * @param {None} 
	 * @return {None} 
	 *
	 */
    logout(){
        ls.clear();
        this.props.closePopup();
    }

    /**
     * Render the profile popup.
     * @param {None}
     * @return {html} Returns a html block of profile component. 
     */
    render() {
        const data = ["email", "reminder"];

        return (
            <div className='profile' onClick={this.props.closePopup}>
                <div className='profile_inner' onClick={e => e.stopPropagation()}>
                    <div className='top'>
                        <span>
                        <div>
                            <div className='profile_icon'>
                                <Icon iconName='times' onClick={this.props.closePopup} />
                            </div>
        
                            <div className='profile_title'>{this.state.title}</div>
                        </div>
                        </span>
                    </div>
                    <div className='middle'>
                        {data.map(key => {
                            if (key === 'email') {
                                return (
                                    <ProfileRow 
                                         key={key} 
                                         field={key}
                                         iconName={config.icons[key]}
                                         placeholder={"Enter Email Address."}
                                         details={this.state[key]}
                                         editing={this.state.editing}
                                         handleEditResult={this.handleEditResult.bind(this)} 
                                    />
                                );
                            }
                            
                            else {
                                return (
                                    <ProfileRow 
                                         key={key} 
                                         field={key}
                                         iconName={config.icons[key]}
                                         placeholder={"Enter default reminder time."}
                                         details={this.state[key]}
                                         editing={this.state.editing}
                                         handleEditResult={this.handleEditResult.bind(this)} 
                                    />
                                );
                            }
                        })}
                    
                        { this.state.validEmail ?
                            (this.state.validReminder ?
                                <span></span> : <span style={{color:'red'}}>Invalid Reminder Time!</span>
                            ): (this.state.validReminder ?
                                <span style={{color:'red'}}>Invalid Email Address!</span> : <span style={{color:'red'}}>Invalid Email and Reminder Time!</span>)
                        }
                        
                    </div>
                    <div className='bottom'>
                        <div className='left'>
                            { this.state.editing ? 
                            (<Icon iconName='save' onClick={this.handleSubmit.bind(this)} />)
                            : (<Icon iconName='pen' onClick={this.changeEditingState.bind(this)}/>)
                            }
                            
                        </div>
                        <div className='right'>
                            <Link to="/" style={{textDecoration: 'none', color:'black'}}>
                                <Icon style={{color:'black'}} iconName='sign-out-alt' onClick={() => this.logout()}/>
                            </Link>
                        </div>
                    </div>
                </div>
            </div>
        ); 
    }
}

function mapStateToProps(state){
    return {
        user:state.auth,
        userProfile:state.userProfile
    }
}

export default connect(mapStateToProps,actions)(Profile);



