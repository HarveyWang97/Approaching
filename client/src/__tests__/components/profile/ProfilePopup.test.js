import React from 'react';
import mockStore from '../../../__testutils__/mockStore';
import {shallow} from 'enzyme';

import ProfilePopup from '../../../components/profile/ProfilePopup';
import ProfileRow from '../../../components/profile/ProfileRow';
import Icon from '../../../components/common/Icon';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const initialState = {
    userProfile:{
        email:"wang@qq.com",
        notifyTime:"36000000"
    }
};

describe("test the profile popup", () => {
    const wrapper = shallow(<ProfilePopup />,
        {
            context:{store:mockStore(initialState)}   
        });

    it("render as expected",() => {
        expect(wrapper).toMatchSnapshot();
    });

    it("has the email rendered correctly", () => {
        expect(wrapper.dive().find(ProfileRow).at(0).props().details).toEqual("wang@qq.com");
    });

    it("test logout correctly",() => {
        wrapper.dive().find(Icon).at(1).simulate('click');
        
    });
});