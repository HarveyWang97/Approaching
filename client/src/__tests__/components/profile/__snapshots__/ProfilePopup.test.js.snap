// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the profile popup render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Profile) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "deleteEvent": [Function],
      "deleteItem": [Function],
      "fetchEvents": [Function],
      "fetchItems": [Function],
      "fetchProfile": [Function],
      "fetchUser": [Function],
      "insertEvent": [Function],
      "insertItem": [Function],
      "insertUser": [Function],
      "toggleEventSelector": [Function],
      "toggleItemSelector": [Function],
      "togglePictureEditor": [Function],
      "togglePopup": [Function],
      "updateEmail": [Function],
      "updateEvent": [Function],
      "updateItem": [Function],
      "updateNotifyTime": [Function],
      "user": undefined,
      "userProfile": Object {
        "email": "wang@qq.com",
        "notifyTime": "36000000",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "deleteEvent": [Function],
        "deleteItem": [Function],
        "fetchEvents": [Function],
        "fetchItems": [Function],
        "fetchProfile": [Function],
        "fetchUser": [Function],
        "insertEvent": [Function],
        "insertItem": [Function],
        "insertUser": [Function],
        "toggleEventSelector": [Function],
        "toggleItemSelector": [Function],
        "togglePictureEditor": [Function],
        "togglePopup": [Function],
        "updateEmail": [Function],
        "updateEvent": [Function],
        "updateItem": [Function],
        "updateNotifyTime": [Function],
        "user": undefined,
        "userProfile": Object {
          "email": "wang@qq.com",
          "notifyTime": "36000000",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
